app/models/jobstatusdata.rb

```diff
@@ -118,6 +120,43 @@ def extended_data_slurm(info)
     self
   end
 
+  # Store additional data about the job. (PBSPro-specific)
+  #
+  # Parses the `native` info function for additional information about jobs on PBSPRO systems.
+  #
+  # @return [Jobstatusdata] self
+  def extended_data_pbspro(info)
+    return unless info.native
+    self.account = info.group_list || ''
+
+    attributes = []
+    attributes.push Attribute.new "Cluster", self.cluster_title
+    attributes.push Attribute.new "PBS Id", self.pbsid
+    attributes.push Attribute.new "Job Name", self.jobname
+    attributes.push Attribute.new "User", self.username
+    attributes.push Attribute.new "Account", self.account
+    attributes.push Attribute.new "Walltime", (info.native.fetch(:Resource_List, {})[:walltime].presence || "00:00:00")
+    attributes.push Attribute.new "Walltime Used", info.native.fetch(:resources_used, {})[:walltime].presence || '0'
+    node_count = info.native.fetch(:Resource_List, {})[:nodect].to_i
+    attributes.push Attribute.new "Node Count", node_count
+    ppn = info.native[:Resource_List][:ncpus].presence || '0'
```

Comment:
You may want to use `fetch(:Resource_List, {})` to avoid any possible crashes here.